### DISCUSSION TITLE
Security Course | PasswordStore Audit - Missing Libarires after cloning

### DISCUSSION THREAD

**Discussion by operation-c** [Cyfrin/security-and-auditing-full-course-s23#237] [Q&A]
This write-up aims to provide clarity for anyone experiencing issues after cloning the password repository, specifically when the folders inside the lib directory appear empty: 

After cloning, enter `forge compile` 

If you run into any issues while compiling such as,
**Issue 1**:
```
error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL (err 8)
error: 3151 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
fatal: clone of 'https://github.com/smartcontractkit/chainlink-brownie-contracts' into submodule path '/home/<USER>/foundry-24/cyfrin-contract-security/passwordstore-security-review/3-passwordstore-audit/lib/chainlink-brownie-contracts' failed
Failed to clone 'lib/chainlink-brownie-contracts'. Retry scheduled 
Your project has missing dependencies that could not be installed.
```
OR 

**Issue 2**:
```
Error: 
Compiler run failed:
Error (6275): Source "forge-std/Script.sol" not found: File not found. Searched the following locations: "/home/<USER>/foundry-24/cyfrin-contract-security/passwordstore-security-review/3-passwordstore-audit".
ParserError: Source "forge-std/Script.sol" not found: File not found. Searched the following locations: "/home/<USER>/foundry-24/cyfrin-contract-security/passwordstore-security-review/3-passwordstore-audit".
 --> script/DeployPasswordStore.s.sol:4:1:
 ```

To resolve `issue 1`, enter the following cmd: `git config --global http.postBuffer 500000000`. This will increase the buffer size to handle larger files. Additionally, if you're using a VPN with multi-hop, DAITA, or a custom MTU, it's best to disable these features for the download. Instead, proceed with a single point of entry and exit. 

To resolve `issue 2`, go to the `.gitmodules` file and make sure you have the following layout:
```
[submodule "lib/foundry-devops"]
	path = lib/foundry-devops
	url = https://github.com/chainaccelorg/foundry-devops
[submodule "lib/chainlink-brownie-contracts"]
	path = lib/chainlink-brownie-contracts
	url = https://github.com/smartcontractkit/chainlink-brownie-contracts
[submodule "lib/forge-std"]
	path = lib/forge-std
	url = https://github.com/foundry-rs/forge-std
[submodule "lib/openzeppelin-contracts"]
	path = lib/openzeppelin-contracts
	url = https://github.com/openzeppelin/openzeppelin-contracts
```
After completing the previous step, enter the following cmd: `git submodule update --init --recursive`. This will initialize all submodules, effectively cloning the necessary missing modules. As a result, folders such as forge-std, foundry-devops, etc.. will no longer be empty and you will be able to successfully compile/ build.

**Comment by EngrPips** [comment]
Interesting, I am glad you figured that out. Thanks for the heads-up too.
